### PR TITLE
Displaying review status for individual teammates/groups in review stages

### PR DESCRIPTION
**Pull request by e-kolpakov**
**Description:** This PR adds displaying review status for teammates/groups in teammate/group selector
**JIRA:** [MCKIN-3478](https://edx-wiki.atlassian.net/browse/MCKIN-3478)
**Testing instructions:**
0. Configure GPv2
1. Log in as student in a group with at least one other student; this student should not have completed peer grading and team evaluation reviews.
2. Open Team Evaluation stage (if there are multiple such stages - any would do just fine).
3. Note that Teammate selector now displays review state icons. 
4. Submit review; note that status icon have changed (also might make sense to refresh the page and make sure it is persisted)
5. Open Peer Grading stage.
6. Note that Group Selector now displays review state icons
7. Submit review; note that status icon have changed (also might make sense to refresh the page and make sure it is persisted)

### CODE REVIEW

File: group_project_v2/project_api/dtos.py [hunk 1]
```diff
@@ -24,7 +34,7 @@ def __init__(self, **kwargs):
     def full_name(self):
         if self._full_name:
             return self._full_name
-        return u" ".join([unicode(part) for part in (self.first_name, self.last_name) if part is not None])
+        return super(UserDetails, self).full_name
```
Comment (smarnach) [inline group_project_v2/project_api/dtos.py]: Nit: you might as well move the `_full_name` attribute into the base class.  (It doesn't matter, really, but I like to keep layers of indirection to a minimum.)

File: group_project_v2/public/js/components/review_subject_selector.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if (typeof gettext === "undefined") {
+        window.gettext = function gettext_stub(string) { return string; };
+    }
+
+    var ERROR_REFRESHING_STATUSES = gettext("Error refreshing statuses");
+
+    var get_statuses_endpoint = runtime.handlerUrl(element, "get_statuses");
+    var status_icon_class = ReviewSubjectSelectorConstants.status_icon_class;
+
+    var group_project_dom = $(element).parents(".group-project-xblock-wrapper");
```
Comment (smarnach) [inline group_project_v2/public/js/components/review_subject_selector.js]: This variable name sounds like this represents a DOM object, while it actually is a jQuery object.

File: group_project_v2/public/js/components/review_subject_selector.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    $(document).on(ReviewSubjectSelectorConstants.refresh_statuses_event, function() {
+        resetCssClasses();
+        displaySpinners();
+        $.ajax({
+            dataType: 'json', url: get_statuses_endpoint,  method: 'GET'
+        }).done(function(data){
+            for (var review_subject_id in data) {
+                if (!data.hasOwnProperty(review_subject_id)) {
```
Comment (smarnach) [inline group_project_v2/public/js/components/review_subject_selector.js]: Why is this necessary?  Can this be avoided by using `$.each()` instead of the for loop?

Comment (e-kolpakov) [inline group_project_v2/public/js/components/review_subject_selector.js]: @smarnach no, as  (1) `data` is not a jQuery object, so no point using `$.each` (I would use underscore's `_.keys` if it was available, but underscore is not there) and (2) that's how iterating over dictionary is supposed to be done in pure JS (i.e. without helper libs).

Comment (smarnach) [inline group_project_v2/public/js/components/review_subject_selector.js]: @e-kolpakov I mainly meant to understand what properties you need to filter out here.

For what it's worth, `$.each()` can be used to iterate over JS objects and arrays, not only jQuery objects.

File: tests/js/karma.conf.js [hunk 4]
```diff
@@ -91,6 +91,6 @@ module.exports = function(config) {
 
     // Continuous Integration mode
     // if true, Karma captures browsers, runs the tests and exits
-    singleRun: true
+    singleRun: false
```
Comment (smarnach) [inline tests/js/karma.conf.js]: Why this change?  Shouldn't this be true when running the Karma tests on Travis?

File: group_project_v2/public/js/components/review_subject_selector.js (hunk 3 above)
Comment (e-kolpakov) [inline group_project_v2/public/js/components/review_subject_selector.js]: @smarnach basically `data` is a dict of `review_subject_id` -> `review_state`, and I don't need anything else here.

IMHO, using `$.each` on arrays/objects is at best is as good as `hasOwnProperty` check: it needs an iteratee function, so are prone to all kinds of closure-over-mutable and `this` mistakes. So, in cases like this I err on `hasOwnProperty` side for the sake of simplicity, as it is clearly synchronous and no closures are created.

File: tests/js/karma.conf.js (hunk 4 above)
Comment (e-kolpakov) [inline tests/js/karma.conf.js]: @smarnach CI build was changed to pass `--single-run` command line parameter (takes precedence over conf file) in [this commit](https://github.com/open-craft/xblock-group-project-v2/commit/16553537b3a4aa36be062dae1103482958bb94f7). Basically, this change should be the part of that commit, but I missed it than.

File: group_project_v2/public/js/components/review_subject_selector.js (hunk 3 above)
Comment (smarnach) [inline group_project_v2/public/js/components/review_subject_selector.js]: @e-kolpakov Fair enough.  I still don't quite get why `hasOwnProperty` is needed here, since my understanding is that you would only get the elements of the mapping you want even without the check, but it doesn't hurt in any case.

File: tests/js/karma.conf.js (hunk 4 above)
Comment (smarnach) [inline tests/js/karma.conf.js]: @e-kolpakov Makes sense, thanks.

**Comment by smarnach** [comment]
:+1:
